src/serde/de_br.rs

```diff
@@ -21,6 +21,47 @@ pub fn node_from_stream_backrefs(
     allocator: &mut Allocator,
     f: &mut Cursor<&[u8]>,
     mut backref_callback: impl FnMut(NodePtr),
+) -> io::Result<NodePtr> {
+    let mut values = Vec::<(NodePtr, Option<NodePtr>)>::new();
+    let mut ops = vec![ParseOp::SExp];
+
+    let mut b = [0; 1];
+    while let Some(op) = ops.pop() {
+        match op {
+            ParseOp::SExp => {
+                f.read_exact(&mut b)?;
+                if b[0] == CONS_BOX_MARKER {
+                    ops.push(ParseOp::Cons);
+                    ops.push(ParseOp::SExp);
+                    ops.push(ParseOp::SExp);
+                } else if b[0] == BACK_REFERENCE {
+                    let path = parse_path(f)?;
+                    let back_reference = traverse_path_with_vec(allocator, path, &mut values)?;
+                    backref_callback(back_reference);
+                    values.push((back_reference, None));
+                } else {
+                    let new_atom = parse_atom(allocator, b[0], f)?;
+                    values.push((new_atom, None));
+                }
+            }
+            ParseOp::Cons => {
+                // cons
+                // pop left and right values off of the "values" stack, then
+                // push the new pair onto it
+                let right = values.pop().expect("No cons without two vals.");
+                let left = values.pop().expect("No cons without two vals.");
+                let root_node = allocator.new_pair(left.0, right.0)?;
+                values.push((root_node, None));
```

Comment:
the allocator must be informed, somehow, that we're pretending to create a pair here, for the purpose of the allocator limits. One option could be to decrement the limit by one. Then, if `traverse_path_with_vec()` has to create some of the pairs to make up the stack, you would have to increment the limit again, for each pair created.

This is important in order to not change consensus rules between the old and the new function.

Reply:
you need this call here too:
```
allocator.reduce_pair_max(1)?;
```

Reply:
Done